workshops/models.py

```diff
@@ -296,17 +296,20 @@ def get_by_ident(ident):
         YYYY-MM-DD-site slug, and indicates an event whose dates are firm.
         The real indicator is the 'published' flag.
         '''
-
-        # match event ID
-        if re.match(r'^\d+$', ident):
-            return Event.objects.get(id=ident)
-
-        # match event slug in format YYYY-MM-**** (ie. this works for
-        # YYYY-MM-DD-**** too)
-        if re.match(r'^\d{4}-\d{2}-.+$', ident):
-            return Event.objects.get(slug=ident)
-
-        raise ObjectDoesNotExist(ident)
+        kwargs = {}
+        try:
+            # first try to match using ID
+            kwargs["id"] = int(ident)
+        except ValueError:
+            # match event slug in format YYYY-MM-****
+            #(ie. this works for YYYY-MM-DD-**** too)
+            if re.match(r'^\d{4}-\d{2}-.+$', ident):
+                kwargs["slug"] = ident
+            else:
+                raise ObjectDoesNotExist(ident)
+
+        # may throw DoesNotExist exception
+        return Event.objects.get(**kwargs)
```

Comment:
On Mon, Apr 20, 2015 at 01:20:57PM -0700, Piotr Banaszkiewicz wrote:
> No, please don't get rid of `EventForm.clean_slug()`.

I think it's good to encourage users to use YYYY-MM-DD-SITE slugs, but
I see no reason to *require* it.  So I'd just drop a note to that
effect in the help_text for Event.slug, drop the clean_slug addition,
and use the int-fallback-to-slug approach when interpreting URLs.

I think folks who use integer slugs that collide with event IDs
deserve any suffering they get ;), but if you want to check to make
sure that the slug isn't an integer, I'm ok with that.